src/sinks/influxdb/logs.rs

```diff
@@ -0,0 +1,657 @@
+            .filter(|(key, _)| !tag_keys.contains(key))
+            .map(|(key, value)| (key, value.to_field()))
+            .collect();
+
+        influx_line_protocol(
+            measurement,
+            "logs",
+            Some(tags),
+            Some(fields),
+            timestamp,
+            &mut output,
+        );
+
+        Some(output.into_bytes())
+    }
+
+    fn build_request(&self, events: Self::Output) -> http::Request<Vec<u8>> {
+        let settings = influxdb_settings(
+            self.influxdb1_settings.clone(),
+            self.influxdb2_settings.clone(),
+        )
+        .unwrap();
+
+        let endpoint = self.endpoint.clone();
+        let token = settings.token();
+
+        let uri = settings.write_uri(endpoint).unwrap();
+
+        let mut builder = Request::builder();
+        builder.method(Method::POST);
+        builder.uri(uri.clone());
+
+        builder.header("Content-Type", "text/plain");
+        builder.header("Authorization", format!("Token {}", token));
+        builder.body(events).unwrap()
+    }
+}
+
+impl Value {
+    pub fn to_field(&self) -> Field {
+        match self {
+            Value::Integer(num) => Field::Int(*num),
+            Value::Float(num) => Field::Float(*num),
+            Value::Boolean(b) => Field::Bool(*b),
+            _ => Field::String(self.to_string_lossy()),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::event::Event;
+    use crate::sinks::influxdb::test_util::{assert_fields, split_line_protocol, ts};
+    use crate::sinks::util::http::HttpSink;
+    use crate::sinks::util::test::build_test_server;
+    use crate::test_util;
+    use chrono::offset::TimeZone;
+    use chrono::Utc;
+    use futures01::{Sink, Stream};
+
+    #[test]
+    fn test_config_without_tags() {
+        let config = r#"
+            namespace = "vector-logs"
+            endpoint = "http://localhost:9999"
+            bucket = "my-bucket"
+            org = "my-org"
+            token = "my-token"
+        "#;
+
+        let _: InfluxDBLogsConfig = toml::from_str(&config).unwrap();
+    }
+
+    #[test]
+    fn test_encode_event() {
+        let mut event = Event::from("hello");
+        event.as_mut_log().insert("host", "aws.cloud.eur");
+        event.as_mut_log().insert("source_type", "file");
+
+        event.as_mut_log().insert("int", 4);
+        event.as_mut_log().insert("float", 5.5);
+        event.as_mut_log().insert("bool", true);
+        event.as_mut_log().insert("string", "thisisastring");
+        event.as_mut_log().insert("timestamp", ts());
+
+        let (config, _, _) = crate::sinks::util::test::load_sink::<InfluxDBLogsConfig>(
+            r#"
+            namespace = "ns"
+            endpoint = "http://localhost:9999"
+            bucket = "my-bucket"
+            org = "my-org"
+            token = "my-token"
+        "#,
+        )
+        .unwrap();
+
+        let bytes = config.encode_event(event.clone()).unwrap();
```

Comment:
Unnecessary `clone` (not that it matters much in a test)

Reply:
Done